engine/compiler/util.go

```diff
@@ -190,6 +172,15 @@ func min(a, b int64) int64 {
 	return b
 }
 
+func first(values ...int64) int64 {
```

Comment:
can you rename the function to describe more accurately what it does. ie this function returns first non zero integer